src/role.builder.go

```diff
@@ -0,0 +1,33 @@
+package main
+
+import . "screepsgo/src/screeps"
+
+func roleBuilder(creep Creep) {
+	if creep.Memory().Get("building").Truthy() && *creep.Store().GetUsedCapacity(&RESOURCE_ENERGY) == 0 {

```

Comment:
`creep.Memory().Get("building").Truthy()` also looks like a good candidate for abstraction.